engine/order_manager.go

```diff
@@ -601,22 +623,72 @@ func (m *OrderManager) processOrders() {
 
 			for z := range result {
 				ord := &result[z]
-				result := m.orderStore.add(ord)
-				if result != ErrOrdersAlreadyExists {
-					msg := fmt.Sprintf("Order manager: Exchange %s added order ID=%v pair=%v price=%v amount=%v side=%v type=%v.",
-						ord.Exchange, ord.ID, ord.Pair, ord.Price, ord.Amount, ord.Side, ord.Type)
+				if !m.Exists(ord) {
+					err = m.Add(ord)
+					if err != nil {
+						log.Errorf(log.OrderMgr,
+							"Order manager: Exchange %s unable to add order ID=%v internal ID=%v pair=%v price=%.8f amount=%.8f side=%v type=%v: %s",
+							ord.Exchange, ord.ID, ord.InternalOrderID, ord.Pair, ord.Price, ord.Amount, ord.Side, ord.Type, err)
+						continue
+					}
+					msg := fmt.Sprintf("Order manager: Exchange %s added order ID=%v internal ID=%v pair=%v price=%.8f amount=%.8f side=%v type=%v.",
+						ord.Exchange, ord.ID, ord.InternalOrderID, ord.Pair, ord.Price, ord.Amount, ord.Side, ord.Type)
 					log.Debugf(log.OrderMgr, "%v", msg)
 					m.orderStore.commsManager.PushEvent(base.Event{
 						Type:    "order",
 						Message: msg,
 					})
+					checkedOrderIDs[ord.InternalOrderID] = true
 					continue
+				} else {
+					err = m.UpdateExistingOrder(ord)
+					if err != nil {
+						log.Errorf(log.OrderMgr,
+							"Order manager: Exchange %s unable to update order ID=%v internal ID=%v pair=%v price=%.8f amount=%.8f side=%v type=%v: %s",
+							ord.Exchange, ord.ID, ord.InternalOrderID, ord.Pair, ord.Price, ord.Amount, ord.Side, ord.Type, err)
+						continue
+					}
+					ord, _ = m.GetByExchangeAndID(ord.Exchange, ord.ID)
+					checkedOrderIDs[ord.InternalOrderID] = true
+					continue
+				}
+			}
+			if !exchanges[i].GetBase().GetSupportedFeatures().RESTCapabilities.GetOrder {
+				continue
+			}
+			for x := range ordersToCheck {
+				curTime := time.Now()
+				// Only check if at least 1 minute not updated
+				if curTime.Sub(ordersToCheck[x].LastUpdated) < time.Minute {
+					checkedOrderIDs[ordersToCheck[x].InternalOrderID] = true
+				} else if checkedOrderIDs[ordersToCheck[x].InternalOrderID] == false {
+					log.Debugf(log.OrderMgr, "Order manager: Exchange %s Unchecked order ID=%v internal ID=%v pair=%v price=%.8f amount=%.8f side=%v type=%v",
+						ordersToCheck[x].Exchange, ordersToCheck[x].ID, ordersToCheck[x].InternalOrderID, ordersToCheck[x].Pair, ordersToCheck[x].Price,
+						ordersToCheck[x].Amount, ordersToCheck[x].Side, ordersToCheck[x].Type)
+					go m.FetchAndUpdateExchangeOrder(exchanges[i], &ordersToCheck[x], supportedAssets[y], curTime)
```

Comment:
This may cause some issues when we have ^1000 limit orders. This *might* bunch everything up at the rate limiter. I think we could employ a more linear approach: 

```go
			go func() {
				for x := range ordersToCheck {
					curTime := time.Now()
					// Only check if at least 1 minute not updated
					if curTime.Sub(ordersToCheck[x].LastUpdated) < time.Minute {
						continue
					}

					if checkedOrderIDs[ordersToCheck[x].InternalOrderID] {
						continue
					}

					log.Debugf(log.OrderMgr, "Order manager: Exchange %s Unchecked order ID=%v internal ID=%v pair=%v price=%.8f amount=%.8f side=%v type=%v",
						ordersToCheck[x].Exchange, ordersToCheck[x].ID, ordersToCheck[x].InternalOrderID, ordersToCheck[x].Pair, ordersToCheck[x].Price,
						ordersToCheck[x].Amount, ordersToCheck[x].Side, ordersToCheck[x].Type)
					m.FetchAndUpdateExchangeOrder(exchanges[i], &ordersToCheck[x], supportedAssets[y], curTime)
				}
			}()
```

and then maybe add a waitgroup to processOrders() just so we don't exceed order manager delay with our HTTP requests. Let me know what you think.

Reply:
Works for me!
This could create an issue if say 1000 orders are cancelled; if you want I can set up a waitgroup for the current solution (thanks @gloriousCode !)